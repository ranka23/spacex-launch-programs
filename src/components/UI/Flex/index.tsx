import React, { ReactNode } from "react";

import styles from "./styles.module.scss";

interface FlexProps {
  className?: string;
  children: ReactNode;
}

const Flex: React.FC<FlexProps> = ({ children, className }: FlexProps) => {
  return (
    <div data-test="flex" className={`${styles.Flex} ${className}`}>
      {children}
    </div>
  );
};

Flex.defaultProps = {
  className: "",
};

export default Flex;
