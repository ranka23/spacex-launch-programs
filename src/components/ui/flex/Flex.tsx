/* eslint-disable react/require-default-props */
import React from "react";
import styles from "./flex.module.scss";

interface FlexProps {
  className?: string;
  children: JSX.Element[];
}

const Flex: React.FC<FlexProps> = ({ children, className }: FlexProps) => {
  return <div className={`${styles.flex} ${className}`}>{children}</div>;
};
export default Flex;
