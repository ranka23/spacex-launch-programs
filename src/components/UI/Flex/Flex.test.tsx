import React, { ReactNode } from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Flex from "./index";

describe("<Flex />", () => {
  const tree = (children?: ReactNode) =>
    render(
      <MemoryRouter>
        <Flex>{children}</Flex>
      </MemoryRouter>
    ).container.firstChild;

  it("renders nothing if no children", () => {
    expect(tree()).toMatchSnapshot();
  });

  it("renders children if no errors", () => {
    expect(
      tree(
        <div>
          <h1>Hello World</h1>
        </div>
      )
    ).toMatchSnapshot();
  });
});
