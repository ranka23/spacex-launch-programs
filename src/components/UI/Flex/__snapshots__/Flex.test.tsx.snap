// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders children if no errors 1`] = `
<div
  class="Flex "
  data-test="flex"
>
  <div>
    <h1>
      Hello World
    </h1>
  </div>
</div>
`;

exports[`<Flex /> renders nothing if no children 1`] = `
<div
  class="Flex "
  data-test="flex"
/>
`;
